app/models/miq_alert.rb

```diff
@@ -89,6 +103,12 @@ def validate_automate_expressions
     valid
   end
 
+  def validate_single_expression
+    unless miq_expression.nil? || hash_expression.nil?
+      errors.add("Alert", "must not have both miq_expression and hash_expression set")
+    end
```

Comment:
:+1: 
It's worth validating that there is exactly one value. That would allow simplifying some code in this PR
(line 555 for example)

Reply:
Validating that there is exactly one value means that you will not be able to have a `nil` expression which was possible until now, so I preferred not to change that behavior.